admin_guide/cluster_management.rst

```diff
@@ -28,12 +28,105 @@ To add a new node to the cluster, you first need to add the DNS name or IP addre
+  that appropriate rows to update or delete can be identified on the
+  subscriber side. By default, this is the primary key, if there is
+  one. Another unique index (with certain additional requirements) can
+  also be set to be the replica identity.
+
+In other words, if your distributed table has a primary key defined then it's ready for shard rebalancing with no extra work. However if it doesn't have a primary key or an explicitly defined replica identity, then attempting to rebalance it will cause an error. For instance:
+
+.. code-block:: sql
+
+  -- creating the following table without REPLICA IDENTITY or PRIMARY KEY
+  CREATE TABLE test_table (key int not null, value text not null);
+  SELECT create_distributed_table('test_table', 'key');
+
+  -- running shard rebalancer with default behavior
+  SELECT rebalance_table_shards('test_table');
+
+  /*
+  NOTICE:  Moving shard 102040 from localhost:9701 to localhost:9700 ...
+  ERROR: cannot use logical replication to transfer shards of the
+    relation test_table since it doesn't have a REPLICA IDENTITY or
+    PRIMARY KEY
+  DETAIL:  UPDATE and DELETE commands on the shard will error out during
+    logical replication unless there is a REPLICA IDENTIY or PRIMARY KEY.
+  HINT:  If you wish to continue without a replica identity set the
+    shard_transfer_mode to 'force_logical' or 'block_writes'.
+  */
+
+There are four solutions to this error.
+
+**Solution 1, add primary key**
+
+Add a primary key to the table. If the desired key happens to be the distribution column, then it's quite easy, just add the constraint. Otherwise, a primary key with a non-distribution column must be composite and contain the distribution column too.
+
+**Solution 2, add replica identity from index**
+
+Create a unique index on a column, and use that for a replica identity. As in option one, the index must cover the distribution column. From our previous example:
+
+.. code-block:: sql
 
+  -- we're indexing value, but the index must cover the
+  -- distribution column too
+  CREATE UNIQUE INDEX test_value_unique
+    ON test_table (value, key);
+
+  -- now set the replica identity to use the index
+  ALTER TABLE test_table REPLICA IDENTITY
+    USING INDEX test_value_unique;
+
+This example would definitely fix the error with ``rebalance_table_shards`` but it's unrealistic insofar as the index includes every column in the table. In a wider table you want to restrict this index to as few columns as you can, which helps replication performance.
+
+.. note::
+
+  While ``REPLICA IDENTITY USING INDEX`` is fine, we recommend **against** adding ``REPLICA IDENTITY FULL`` to a table. This setting would result in each update/delete doing a full-table-scan on the subscriber side to find the tuple with those rows. In our testing we’ve found this to result in worse performance than even solution four below.
+
+**Solution 3, force logical replication (on append-only tables)**
```

Comment:
Can we rename this to maybe "Insert-only"? We have used append-only tables in the context of append-partitioning, and so wanted to keep those a bit separate.